plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/tree/library/JarFileNode.java

```diff
@@ -145,7 +145,7 @@ public boolean isLeaf() {
 
   @Override
   public Path getLocation() {
-    return Path.valueOf(getData().getUri());
+    return Path.valueOf(getData().getPath());
```

Comment:
Che IDE uses VirtualFile#getPath() as an identifier for the virtual file. However, when we get things via LSP (like "textDocument/references"), we only get an URI. So the URI is the only unique identifier we have for a file (possibly a classfile). So we need to get the URI into the Path somehow. 
There seems to have been a concrete case where the old version did not work. Can you say what that case was?

Reply:
Yes, I missed case with "textDocument/references". Thanks for spotting!